assets/app/views/browse/edit-build-config.html

```diff
@@ -0,0 +1,257 @@
+                                id="sourceContextDir"
+                                name="sourceContextDir"
+                                type="text"
+                                ng-model="updatedBuildConfig.spec.source.git.contextDir"
+                                aria-describedby="from_source_help"
+                                placeholder="/">
+                            </div>
+                          </div>
+                        </div>
+
+                        <div class="form-group">
+                          <div class="checkbox" ng-if="buildConfig.spec.strategy.type === 'Docker'">
+                            <label>
+                              <input type="checkbox" ng-model="options.noCache"/>
+                              Execute the docker build with the --no-cache=true flag
+                            </label>
+                          </div>
+                        </div>
+
+                        <div class="form-group">
+                          <label for="buildFrom">Build From: </label>
+                          <select 
+                            class="form-control simpleselector"
+                            ng-model="options.pickedBuildFromType"
+                            ng-options="type.id as type.title for type in buildFromTypes">
+                          </select>
+                        </div>
+
+                        <div ng-if="options.pickedBuildFromType=='ImageStreamTag'" class="form-group">
+                          <label for="builderImage">Builder Image</label>
+                          <div class="builder-image-selects multiselectors">
+                            <div class="label-with-selector">
+                              <label class="multiselector-label">Namespace</label>
+                              <select 
+                                class="form-control multiselector select-builder-namespace"
+                                ng-model="options.pickedBuildFromNamespace"
+                                ng-options="name for name in buildFrom.projects track by name"
+                                ng-change="updateImageStreams(options.pickedBuildFromNamespace, true, false)">
+                              </select>
+                            </div>
+                            <span class="punctuation">/</span>
+                            <div class="label-with-selector">
+                              <label class="multiselector-label">Image Stream</label>
+                              <select
+                                class="form-control multiselector select-builder-name"
+                                ng-model="options.pickedBuildFromImageStream"
+                                ng-options="name as name for name in buildFrom.imageStreams track by name"
+                                ng-change="updateTags(options.pickedBuildFromImageStream, true)">
+                              </select>
```

Comment:
You should be able to type in one that doesn't exist (I'm assuming that was
implemented, havent tried it), so don't make it an error, make it a warning.

On Tue, Jan 5, 2016 at 1:04 PM, Sam Padgett <notifications@github.com>
wrote:

> In assets/app/views/browse/edit-build-config.html
> <https://github.com/openshift/origin/pull/6528#discussion_r48874909>:
>
> > +                              <select
> > +                                class="form-control multiselector select-builder-namespace"
> > +                                ng-model="options.pickedBuildFromNamespace"
> > +                                ng-options="name for name in buildFrom.projects track by name"
> > +                                ng-change="updateImageStreams(options.pickedBuildFromNamespace, true, false)">
> > +                              </select>
> > +                            </div>
> > +                            <span class="punctuation">/</span>
> > +                            <div class="label-with-selector">
> > +                              <label class="multiselector-label">Image Stream</label>
> > +                              <select
> > +                                class="form-control multiselector select-builder-name"
> > +                                ng-model="options.pickedBuildFromImageStream"
> > +                                ng-options="name as name for name in buildFrom.imageStreams track by name"
> > +                                ng-change="updateTags(options.pickedBuildFromImageStream, true)">
> > +                              </select>
>
> I think we need an error message below saying there are no image streams
> for the namespace when the list is empty.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/6528/files#r48874909>.
>